src/Asset/TagRenderer.php

```diff
@@ -47,24 +55,43 @@ public function __construct(
 
         $this->packages = $packages;
         $this->defaultAttributes = $defaultAttributes;
+        $this->defaultScriptAttributes = $defaultScriptAttributes;
+        $this->defaultLinkAttributes = $defaultLinkAttributes;
 
         $this->reset();
     }
 
-    public function renderWebpackScriptTags(string $entryName, string $packageName = null, string $entrypointName = '_default'): string
+    public function setEventDispatcher(EventDispatcherInterface $eventDispatcher)
```

Comment:
Why using setter injection instead of an optional constructor argument?

Reply:
it was a subjective call... it felt odd to put a service argument after 3 optional array args.